File: springboot/src/main/java/com/example/crowdsourcing/dao/bean/OffineOrder.java
package com.example.crowdsourcing.dao.bean;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;

@Entity
public class OffineOrder {


    @Id
    @GeneratedValue
    private int id;


    @Column
    int peopleId;
    @Column
    private String title, limitedTime;
    @Column(name = "des")
    String describe;
    @Column(name = "requir")
    String require;

    // 0 代表无限制    1代表安卓 2代表苹果
    @Column
    int platFormLimit;


    @Column
    String buyMessages;







    @Column
    int end;
    @Column
    double price;

    //总数
    @Column
    int total;
    //剩余未领
    @Column
    int remain;
    //等待审核
    @Column
    int  submit;


    //已经完成
    @Column
    int finish;
    //还有一个，就是领了任务还没有做的，那么就是t-r-s-f


    public int getSubmit() {
        return submit;
    }

    public void setSubmit(int submit) {
        this.submit = submit;
    }


    public int getFinish() {
        return finish;
    }

    public void setFinish(int finish) {
        this.finish = finish;
    }


    public int getTotal() {
        return total;
    }

    public void setTotal(int total) {
        this.total = total;
    }

    public int getRemain() {
        return remain;
    }

    public void setRemain(int remain) {
        this.remain = remain;
    }

    public int getEnd() {
        return end;
    }

    public void setEnd(int end) {
        this.end = end;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getPeopleId() {
        return peopleId;
    }

    public void setPeopleId(int peopleId) {
        this.peopleId = peopleId;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDescribe() {
        return describe;
    }

    public void setDescribe(String describe) {
        this.describe = describe;
    }

    public String getRequire() {
        return require;
    }

    public void setRequire(String require) {
        this.require = require;
    }

    public String getLimitedTime() {
        return limitedTime;
    }

    public void setLimitedTime(String limitedTime) {
        this.limitedTime = limitedTime;
    }

    public int getPlatFormLimit() {
        return platFormLimit;
    }

    public void setPlatFormLimit(int platFormLimit) {
        this.platFormLimit = platFormLimit;
    }

    public String getBuyMessages() {
        return buyMessages;
    }

    public void setBuyMessages(String buyMessages) {
        this.buyMessages = buyMessages;
    }

    public double getPrice() {
        return price;
    }

    public void setPrice(double price) {
        this.price = price;
    }

    public void remainMinus(){
        remain--;
    }

    public void sumbitR(){
        finish++;
    }


}

File: flutter/crowdsourcing/lib/models/object/order/offine/OffineOrder.dart
import 'dart:convert';

import 'package:crowdsourcing/models/object/order/offine/BuyMessage.dart';
import 'package:crowdsourcing/models/object/order/Order.dart';
import 'package:crowdsourcing/models/object/order/offine/location/Location.dart';

class OffineOrder extends Order {
  List<BuyMessage> buyMessages;
  Location end;

  //double price;

  static const String BUYMESSAGES = 'buyMessages';
  static const String END = "end";

  OffineOrder(
      {String title,
      String describe,
      String require,
      String limitedTime,
      platFormLimit,
        int submit,
      double price,
      int id,
        int finish,
        int total=1,
        int remain ,
      int peopleId,
      this.end,
        DateTime createdTime,
        DateTime finishTime,
        this.buyMessages
      }) {
    this.title = title;
    this.submit =submit??0;
    this.price = price;
    this.describe = describe;
    this.finish=finish??0;
    this.limitedTime = limitedTime;
    this.require = require;
    this.platFormLimit = platFormLimit;
    this.total =total;
    this.remain=remain??total;
    this.id = id;
    this.peopleId = peopleId;
    this.finishTime=finishTime;
    this.createdTime=createdTime;
  }

  static OffineOrder fromJsonMap(Map<String, dynamic> map) {
    if (map == null) return null;
    return OffineOrder(
        title: map[Order.TITLE],
        total: map[Order.TOTAL] ,
        remain: map[Order.REMAIN]?? map[Order.TOTAL],
        describe: map[Order.DESCRIBE],
        require: map[Order.REQUIRE],
        limitedTime: map[Order.LIMITEDTIME],
        submit: map[Order.SUBMIT],
        platFormLimit: map[Order.PLATFORMLIMIT],
        end: Location(id: map[END]),
        buyMessages: ((json.decode(map[BUYMESSAGES])) as List).map<BuyMessage>((it) {
          return BuyMessage.fromJsonMap(it);
        }).toList(),
        peopleId: map[Order.PEOPLEID],
        id: map[Order.ID],
        finish: map[Order.FINISH],
        createdTime:map[Order.CREATEDTIME]!=null? DateTime.fromMicrosecondsSinceEpoch(map[Order.CREATEDTIME]):null,
        finishTime:map[Order.FINISHTIME]!=null? DateTime.fromMicrosecondsSinceEpoch(map[Order.FINISHTIME]):null,
        price: map[Order.Price]);
  }

  Map toJson() {
    return {
      Order.TITLE: title,
      Order.DESCRIBE: describe,
      Order.REQUIRE: require,
      Order.LIMITEDTIME: limitedTime,
      Order.PLATFORMLIMIT: platFormLimit,
      END: end?.id ?? '-1',
      Order.Price: price,
      Order.TOTAL:total,
      Order.REMAIN:remain,
      Order.SUBMIT :submit,
      Order.ID: id,
      Order.FINISH:finish,
      Order.PEOPLEID: peopleId,
      Order.CREATEDTIME:createdTime?.microsecondsSinceEpoch,
      Order.FINISHTIME:finishTime?.microsecondsSinceEpoch,
      BUYMESSAGES: json.encode(buyMessages.map((it) {
        return it.toJson();
      }).toList())
    };
  }
}
